manifests/server/config.pp

```diff
@@ -91,12 +91,48 @@
         quoted   => 'double',
       }
 
-      shellvar { 'slapd_sockets':
-        ensure   => present,
-        target   => '/etc/rc.conf',
-        variable => 'slapd_sockets',
-        value    => join($::openldap::server::ldapi_ifs, ' '),
-        quoted   => 'double',
+      if ($::openldap::server::ldapi_ifs) {
+        shellvar { 'slapd_sockets':
+          ensure   => present,
+          target   => '/etc/rc.conf',
+          variable => 'slapd_sockets',
+          value    => join($::openldap::server::ldapi_ifs, ' '),
+          quoted   => 'double',
+        }
+      }
+
+      if ($openldap::server::provider == 'olc') {
+        # On FreeBSD we need to bootstrap slapd.d
+        $ldif = @(EOL)
+dn: cn=config
+objectClass: olcGlobal
+cn: config
+olcArgsFile: /var/run/openldap/slapd.args
+olcPidFile: /var/run/openldap/slapd.pid
+
+dn: olcDatabase={0}config,cn=config
+objectClass: olcDatabaseConfig
+olcDatabase: {0}config
+olcAccess: to dn.subtree="cn=config" by dn=gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth manage by * none
+
+dn: cn=module,cn=config
+objectClass: olcModuleList
+cn: module
+olcModulepath:  /usr/local/libexec/openldap
+olcModuleload:  back_mdb.la
+olcModuleload:  back_ldap.la
+
+dn: cn=schema,cn=config
+objectClass: olcSchemaConfig
+cn: schema
+
+include: file:///usr/local/etc/openldap/schema/core.ldif
+EOL
+        exec { 'bootstrap cn=config':
+          path    => $::path,
+          command => "echo '${ldif}' | slapadd -n 0 -F ${$::openldap::server::confdir}",
```

Comment:
Redundant `$` and no need for the leading `::` (it was a workaround in the age of Puppet 3)

```suggestion
          command => "echo '${ldif}' | slapadd -n 0 -F ${openldap::server::confdir}",
```